src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConditionalWeakTable.cs

```diff
@@ -157,7 +162,38 @@ public bool Remove(TKey key)
 
             lock (_lock)
             {
-                return _container.Remove(key);
+                return _container.Remove(key, out _);
+            }
+        }
+
+        /// <summary>Removes a key and its value from the table, and returns the removed value if it was present.</summary>
+        /// <param name="key">key to remove. May not be null.</param>
+        /// <param name="value">value removed from the table, if it was present.</param>
+        /// <returns>true if the key is found and removed. Returns false if the key was not in the dictionary.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="key"/> is <see langword="null"/>.</exception>
+        /// <remarks>
+        /// The key may get garbage collected during the <see cref="Remove(TKey, out TValue)"/> operation. If so,
+        /// <see cref="Remove(TKey, out TValue)"/> will not fail or throw, however, the return value can be either
+        /// true or false depending on who wins the race.
+        /// </remarks>
+        public bool Remove(TKey key, [MaybeNullWhen(false)] out TValue value)
+        {
+            if (key is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.key);
+            }
+
+            lock (_lock)
+            {
+                bool found = _container.Remove(key, out object? valueObject);
+
+                // We can safely suppress the nullability warning here, because if we did find the key,
+                // the retrieved value is guaranteed to not be null. The 'Remove' method on the container
+                // doesn't have that annotation just because it's looking for an index, not a boolean,
+                // and there's no way to express something like '[MaybeNullWhen(!= 1)]'.
```

Comment:
This is also not true. It's `MaybeNullWhen(false)` because it can actually be null.